app/src/common/shared/com/igalia/wolvic/ui/widgets/settings/SearchEngineView.java

```diff
@@ -108,6 +108,16 @@ public void onHidden() {
     @Override
     public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
         if (key.equals(getContext().getString(R.string.settings_key_search_engine_id))) {
+            int checkedId = mBinding.searchEngineRadio.getCheckedRadioButtonId();
+            if (checkedId >= 0 && checkedId < mSearchEngines.size()) {
+                SearchEngine selected = mSearchEngines.get(checkedId);
+                String preferenceValue = sharedPreferences.getString(key, "");
```

Comment:
I think `preferenceValue` is too generic. I'd prefer something more descriptive like previousSearchEngineId for example

Reply:
Done.